### PR TITLE
[Woo POS] Implement reader update failure views - retryable, non-retryable, low battery

### LINKED ISSUE #13070: [Woo POS] Implement card reader update SwiftUI views
In #13062 we added SwiftUI views and view models for all the regular connection events, but skipped upgrade views to keep the PR small.

`required` updates are the most commonly applied type, and happen during reader connection. We have no way to know whether they will happen before a connection attempt, so we can't gate the POS experience to exclude them.

We should implement the upgrade progress views, and specific upgrade error views, to match the existing designs.

### PR DESCRIPTION
**Pull request by jaclync**
Closes: #13070 
Currently based off of https://github.com/woocommerce/woocommerce-ios/pull/13085

## Description

This PR implements the SwiftUI views for 3 card reader software update failure cases: retryable, non-retryable, low battery. For some reason, all the `.updatingFailed(tryAgain:...)` instances in the app set the `tryAgain` parameter to `nil`, perhaps we could remove this parameter if we don't plan to implement reader update retry in the foreseeable future.

## Testing information

For easier testing, feel free to mock the error by replacing `CardReaderConnectionController.onInitialization` with

* Non-retryable

```swift
        onUpdateFailed(error: CardReaderServiceError.softwareUpdate(underlyingError: .readerSoftwareUpdateFailed, batteryLevel: 0.2))
```

* Retryable

In addition to the replacement as in the non-retryable case above, also replace the `tryAgain` value to an empty closure:

https://github.com/woocommerce/woocommerce-ios/blob/47e1d4eebb7e5b8ff6c4aee41779538d0e3595c1/WooCommerce/Classes/ViewRelated/CardPresentPayments/CardReaderConnectionController.swift#L615

* Low battery error case

```swift
        onUpdateFailed(error: CardReaderServiceError.softwareUpdate(underlyingError: .readerSoftwareUpdateFailedBatteryLow, batteryLevel: 0.3))
```

And then:

Prerequisite: having a self-hosted store eligible for POS

For each of the 3 error cases with the testing setup above:

- Switch to a store in the prerequisite if needed
- Turn on the feature switch in Menu > Settings > Experimental Features > POS, if needed
- Go to the Menu tab --> the POS row should appear shortly
- Tap to enter POS
- Tap `Connect now` in the bottom bar --> the error modal should be shown

## Screenshots

retryable | non-retryable | low battery
-- | -- | --
Simulator Screenshot - iPad (10th generation) - 2024-06-18 at 14 38 31 | Simulator Screenshot - iPad (10th generation) - 2024-06-18 at 14 34 52 | Simulator Screenshot - iPad (10th generation) - 2024-06-18 at 14 06 55

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-ios/milestone/210">19.2</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/Classes/POS/Card Present Payments/CardPresentPaymentEventDetails.swift [hunk 1]
```diff
@@ -31,7 +31,7 @@ enum CardPresentPaymentEventDetails {
     case updateProgress(requiredUpdate: Bool,
                         progress: Float,
                         cancelUpdate: (() -> Void)?)
-    case updateFailed(tryAgain: (() -> Void)?,
+    case updateFailed(tryAgain: () -> Void,
```
Comment (joshheald) [inline WooCommerce/Classes/POS/Card Present Payments/CardPresentPaymentEventDetails.swift:34]: 👍  Nice improvement. I agree we should remove it throughout if not used in the existing app, along with other stuff we've found like that... but probably outside the direct POS work.

**Review by joshheald — APPROVED** [review]
:shipit: LGTM! Thanks for implementing these.

File: WooCommerce/Classes/POS/Card Present Payments/CardPresentPaymentEventDetails.swift (hunk 1 above)
Comment (jaclync) [reply WooCommerce/Classes/POS/Card Present Payments/CardPresentPaymentEventDetails.swift:34]: I created a tech debt issue to remove two unused parameters found in the reader update work: https://github.com/woocommerce/woocommerce-ios/issues/13095

**Pull request merged**